Realm/ObjectServerTests/run_baas.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/sh
+
+######################################
+#
+# Copyright 2020 Realm Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+######################################
+
+mongodb_url=http://fastdl.mongodb.org/osx/mongodb-osx-ssl-x86_64-4.0.2.tgz;
+transpiler_target=node8-macos;
+server_stitch_lib_url="https://s3.amazonaws.com/mciuploads/mongodb-mongo-master/stitch-support/osx-1010/80f9a13324fc36b2deb400e5a185968f6fa8f64a/stitch-support-4.1.7-319-g80f9a13324.tgz";
+
+function setup_mongod() {
+    curl --silent ${mongodb_url} | tar xz
+    pushd mongodb-*
+    mkdir db_files
+    popd
+}
+
+function run_mongod() {
+    pushd mongodb-*
+    echo "starting mongod..."
+    ./bin/mongod --dbpath ./db_files --port 26000 --replSet test --pidfilepath ./pid.txt
+    popd
+}
+
+function wait_for_mongod() {
+    pushd mongodb-*
+    echo "waiting for mongod to start up"
+    ./bin/mongo --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:26000"); return true}catch(e){return false}}, "timed out connecting")'
+    ./bin/mongo --port 26000 --eval 'rs.initiate()'
+    echo "mongod is up."
+    popd
+}
+
+function setup_stitch() {
+    OG_DIR=`pwd`
+    echo "cloning stitch"
+    git clone git@github.com:10gen/stitch
```

Comment:
We'll want to check out a specific version of stitch rather than just using the latest commit on the default branch. We also don't really want to be cloning, building, and then deleting stitch every time we run the tests; instead it should clone it to a semi-persistent location if it's not already there, or make sure that the correct version is checked out if there is already one there. This could also be done as a build step similar to how the download core build step works rather than at runtime.